src/main/java/au/com/southsky/jfreesane/SaneDevice.java

```diff
@@ -214,4 +217,41 @@ SaneSession getSession() {
   void invalidateOptions() {
     optionTitleMap = null;
   }
+
+  @Override
+  public int hashCode() {
+    int hash = 3;
+    hash = 37 * hash + Objects.hashCode(this.name);
+    hash = 37 * hash + Objects.hashCode(this.vendor);
+    hash = 37 * hash + Objects.hashCode(this.model);
+    hash = 37 * hash + Objects.hashCode(this.type);
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
> Well, you'll recreate (most) of the state on the client, but you need to tell the server as well.

Now upon closer examination of the source, I'm confused because I think the changed values are never stored on the client side. In order to make `SaneDevice` serializable we need to track all changes (eventually only those which are written successfully on the server), because afaik the only access to option values is through `SaneOption` and `SaneOption.get[...]Value` only delegates to `readOption`. There's no option values on the client side, i.e. nothing to serialize.

What do you think about keeping a `Map<SaneOption, Object>` to track (successful) changes of options? A use case would be a (document management) application which allows partial configuration of scanner properties which ought to be persisted across application restarts (e.g. https://github.com/krichter722/document-scanner ;)).